pymc_marketing/mmm/media_transformation.py

```diff
@@ -0,0 +1,278 @@
+        adstock=GeometricAdstock(l_max=15),
+        saturation=HillSaturation(),
+        adstock_first=True,
+    )
+    online_media_transform = MediaTransformation(
+            adstock=GeometricAdstock(l_max=10),
+            saturation=MichaelisMentenSaturation(),
+            adstock_first=False,
+        ),
+    )
+
+Create a media configurations for offline and online media channels:
+
+.. code-block:: python
+
+    from pymc_marketing.mmm import (
+        MediaConfig,
+        MediaConfigs,
+    )
+
+    media_configs: MediaConfigs = [
+        MediaConfig(
+            name="offline",
+            columns=["TV", "Radio"],
+            media_transformation=offline_media_transform,
+        ),
+        MediaConfig(
+            name="online",
+            columns=["Facebook", "Instagram", "YouTube", "TikTok"],
+            media_transformation=online_media_transform,
+        ),
+    ]
+
+
+Apply the media transformation to media data in PyMC model:
+
+.. code-block:: python
+
+    import pymc as pm
+    import pandas as pd
+
+    from pymc_marketing.mmm import get_media_values
+
+    df: pd.DataFrame = ...
+
+    media_columns = get_media_values(media_configs)
+
+    coords = {
+        "date": df["week"],
+        "media": media_columns,
+    }
+    with pm.Model(coords=coords) as model:
+        media_data = pm.Data(
+            "media_data",
+            df.loc[:, media_columns].to_numpy(),
+            dims=("date", "media")
+        )
+        transformed_media_data = apply_media_transformation(
+            media_data,
+            media_configs,
+        )
+
+"""
+
+from dataclasses import dataclass
+
+import pymc as pm
+import pytensor.tensor as pt
+
+from pymc_marketing.mmm.components.adstock import (
+    AdstockTransformation,
+)
+from pymc_marketing.mmm.components.saturation import SaturationTransformation
+
+
+@dataclass
+class MediaTransformation:
+    """Wrapper for applying adstock and saturation transformation to media data.
+
+    Parameters
+    ----------
+    adstock : AdstockTransformation
+        The adstock transformation to apply.
+    saturation : SaturationTransformation
+        The saturation transformation to apply.
+    adstock_first : bool
+        Flag to apply the adstock transformation first.
+
+    Attributes
+    ----------
+    first : AdstockTransformation | SaturationTransformation
+        The first transformation to apply.
+    second : AdstockTransformation | SaturationTransformation
+        The second transformation to apply.
+
+    """
+
+    adstock: AdstockTransformation
+    saturation: SaturationTransformation
+    adstock_first: bool
+
+    def __post_init__(self):
+        """Set the first and second transformations based on the adstock_first flag."""
+        self.first, self.second = (
+            (self.adstock, self.saturation)
+            if self.adstock_first
+            else (self.saturation, self.adstock)
+        )
+
+    def __call__(self, x, dim):
```

Comment:
I use __call__ here instead of an `apply` method used in individual transformations. WYDT?
As mentioned below, we can have 

```python
media_data = ...
online_media_transformation = MediaTransformation(adstock=..., saturation=..., ...)
offline_media_transformation = MediaTransformation(adstock=..., saturation=..., ...)
media_configs = [MediaConfig(media_transformation=online_media_transformation), MediaConfig(media_transformation=offline_media_transformation)]
comb_media_transformation = CombinedMediaTransformation(media_configs=media_configs)

# Make all three work the same
offline_media_transformation(media_data, dims="media")
online_media_transformation(media_data, dims="media")
comb_media_transformation(media_data, dims="media")
```

This would have have many conveniences

Reply:
Nice! 😎